rlclientlib/rl_string_view.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
```

Comment:
This is also available in the vw submodule: `vowpalwabbit/vw_string_view.h` would it make sense to re-use it from there

Reply:
Would it make sense to place `vw_string_view.h` into the `common` dir/lib @jackgerrits ? (see common lib [PR](https://github.com/VowpalWabbit/vowpal_wabbit/pull/2573)

Reply:
We must not include non-public headers out of a submodule. If we're to reuse, it should be from a place like `dir/lib`.

Reply:
> We must not include non-public headers out of a submodule.

Whats the reason behind this?

Reply:
Because it's a tight dependencies between the two repositories in an invisible way.

Since we don't CI or review for external usage of internal headers, the chance of us introducing breakage when we update the vw submodule increases. Introducing friction in the submodule update process is a massive hurdle to our agility.

Once we have something like that _commons_ set of headers in VW that @jackgerrits proposed, we can remove this file.